### PR TITLE
Fixed testRecovery157

### PR DESCRIPTION
**Pull request by CippoX**
Fixing testRecovery157 
#1373

### CODE REVIEW

**Comment by CippoX** [comment]
@ahoppen Let me know if I'm on the right track.

File: Tests/SwiftParserTest/translated/RecoveryTests.swift [hunk 1]
```diff
@@ -1881,16 +1881,17 @@ final class RecoveryTests: XCTestCase {
...
       #"""
-      // <rdar://problem/19833424> QoI: Bad error message when using Objective-C literals (@"Hello") in Swift files
       let myString = 1️⃣@"foo"
       """#,
       diagnostics: [
-        // TODO: Old parser expected error on line 2: string literals in Swift are not preceded by an '@' sign, Fix-It replacements: 16 - 17 = ''
-        DiagnosticSpec(message: "expected expression in variable"),
-        DiagnosticSpec(message: #"extraneous code '@"foo"' at top level"#),
-      ]
+        DiagnosticSpec(message: "string literals in Swift are not preceded by an '@' sign")
```
Comment (kimdv) [inline Tests/SwiftParserTest/translated/RecoveryTests.swift:1890]: Please also add the fix it message to the diagnostic

**Review by kimdv — CHANGES_REQUESTED** [review]

File: Tests/SwiftParserTest/translated/RecoveryTests.swift (hunk 1 above)
Comment (CippoX) [reply Tests/SwiftParserTest/translated/RecoveryTests.swift]: Done 👍

**Review by ahoppen — APPROVED** [review]
Thanks. If all tests parse, I think this is ready to merge as-is. Just mark it as ready for review 😉

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by CippoX** [comment]
This should fix the problem

File: Sources/SwiftParser/Expressions.swift [hunk 2]
```diff
@@ -32,6 +32,8 @@ extension TokenConsumer {
       } else {
         return true
       }
+    case (.primaryExpressionStart(.atSign), let handle)?:
+      break
```
Comment (CippoX) [inline Sources/SwiftParser/Expressions.swift:36]: I've added this lines

**Review by ahoppen — APPROVED** [review]
Thank you. Let’s merge it 👍🏽

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**